### DISCUSSION TITLE
Issue with export on JSON with disjunctions.

### DISCUSSION THREAD

**Discussion by conikeec** [cue-lang/cue#1400] [Q&A]
### Can disjunction be applied on two different types of schemas?
```
https://cuelang.org/play/?id=NDCY9W6wLsX#cue@export@json
```

### What version of CUE are you using (`cue version`)?

<pre>
$ cue version
cue version 0.4.0 darwin/amd64
</pre>

### What did you do?
```
//Platform details (if comprised of many products)
#Platform: {
    #Product,
    products: [...#Product]
}

//Product details
#Product: {
    id: int,
    name: string,
    oneLineDescription?: string,
    detailedDescription?: string
}

product: #Platform | #Product
product: { id : 10, name : "Product", oneLineDescription : "Great Product" }
```

### What did you expect to see?
```
product: { id : 10, name : "Product", oneLineDescription : "Great Product" }
```

### What did you see instead?
```
failed to encode: product: incomplete value {id:10,name:"Product",oneLineDescription:"Great Product",products:[]} | {id:10,name:"Product",oneLineDescription:"Great Product"}
```

**Comment by verdverm** [comment]
By embedding product in platform and then disjuncting them , you have created a subset situation. CUE cannot disambiguate a subset from the superset.

See https://github.com/cue-lang/cue/discussions/1163 for more details

**Comment by conikeec** [comment]
Thanks @verdverm 

From a data modeling standpoint, a platform can be "many products" aggregated 
In cases like these how can we leverage disjunction (`product` or `platform`)?

**Comment by verdverm** [comment]
Long-term: this is the comment to pay attention to: https://github.com/cue-lang/cue/discussions/1163#discussioncomment-1091218

Short-term, you can use a discriminator field. Something that is in the data that can uniquely separate them, as I have used `pid` in this example: https://cuelang.org/play/?id=Oenn0KqbYqK#cue@export@cue

We could refine this depending on desired output and the data characteristics.

@myitcv should we convert this to a discussion?

**Comment by conikeec** [comment]
Thanks again @verdverm for the suggestion. I will try tonight.
Can I update the docs to reflect this use case with an example as I am suspecting this question might be asked again.

**Comment by verdverm** [comment]
You'd have to ask @myitcv that one.

I run https://cuetorials.com and would love to collab on the field subset problem/ example there. I have some thoughts already, if you open a discussion on the repo, we can iterate.

**Comment by conikeec** [comment]
@verdverm https://github.com/cue-lang/cue/discussions/1247

**Comment by mpvl** [comment]
The way to disambiguate in this case is to set a default value:
```
product: #Platform | *#Product
```
This picks the less specific version in case both succeed.

Much older versions of CUE were "smarter" and picked less specific cases over more specific cases. But this was a bit magical, aside from potential runtime inefficiencies, so we require explicit 

I do think there is merit detecting subsumed and thus redundant values in disjunctions in a `vet` command.
